src/group_theory/complement.lean

```diff
@@ -0,0 +1,297 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+
+import group_theory.group_action
+import group_theory.order_of_element
+import group_theory.quotient_group
+
+/-!
+# Complements
+
+In this file we define the complement of a subgroup.
+
+## Main definitions
+
+- `is_complement S T` where `S` and `T` are subsets of `G`
+- `left_transversals S`
+- `left_transversals T`
+
+## Main results
+
+- `is_complement_of_coprime` : Subgroups of coprime order are complements
+- `schur_zassenhaus` : Schur-Zassenhaus for abelian normal subgroups
+-/
+
+open_locale big_operators
+
+namespace subgroup
+
+variables {G : Type*} [group G] (H K : subgroup G) (S T : set G)
+
+/-- `S` and `T` are compelements if `(*) : S × T → G` is a bijection -/
+def is_complement : Prop := ∀ g : G, ∃! x : S × T, x.1.1 * x.2.1 = g
+
+/-- The set of left-complements of `T : set G` -/
+def left_transversals : set (set G) := {S : set G | is_complement S T}
+
+/-- The set of right-complements of `S : set G` -/
+def right_transversals : set (set G) := {T : set G | is_complement S T}
+
+variables {H K S T}
+
+lemma is_complement_iff_bijective :
+  is_complement S T ↔ function.bijective (λ x : S × T, x.1.1 * x.2.1) :=
+(function.bijective_iff_exists_unique _).symm
+
+lemma is_complement.symm (h : is_complement (H : set G) (K : set G)) :
+  is_complement (K : set G) (H : set G) :=
+begin
```

Comment:
Is this statement only true when `K` and `H` are subgroups?

Reply:
I don't think that it's true in general. This proof uses the fact that H and K are closed under inversion.

Reply:
Which lemma below uses that fact? I must be misreading something

Reply:
The function phi uses the inversion operations on H and K